### PR TITLE
#167765381 Setup and Integrate Travis CI with README.md badge

### PR DESCRIPTION
**Pull request by voke1**
### What does this PR do?

- Setup and integrate Travis CI to automatically run test and other builds
- Add Travis CI badge to README.md

### Description of Task to be completed?

- Activate Travis CI on the Repository
- Add .travis.yml configuration file to the codebase
- Include Travis CI badge to README.md

### How should this be manually tested?

- Run git push to trigger build
- Merge pull request to trigger a build

### Any background context you want to provide?

- Currently no tests have been configured, however, test script is made to echo Error and exit with 0.

### What are the relevant pivotal tracker stories?

- #167765381

### Implementation Plan

- https://drive.google.com/file/d/1MMI4EHfbOJc-N24R2gED-TnF_z3Eo2It/view?usp=sharing

### Questions:

### CODE REVIEW

**Review by ugwumadu116 — COMMENTED** [review]
I like the fact that you understand what you are doing and that's great. but for the **.travis.yml**  file the place you are setting up the node version you added stable to be the version `node_js:  - "stable"` this is nice and cool but I think there will be issues in the future if Node js should upgrade to another stable version that will not be compactable with our Travis. Although this may not happen. But I will prefer you be specific with the version. Nodejs version 10 will be cool

**Comment by voke1** [comment]
@ugwumadu116 Thanks for the suggestion. Though, it might unlikely happen, just as you stated. I think is best to be on a safe side. It is noted.

**Review by ugwumadu116 — APPROVED** [review]

**Review by debelistic — APPROVED** [review]

**Review by uggoprince — CHANGES_REQUESTED** [review]
you don't need to include `sudo: false`

where you have ` - "stable"`, replace it with ` - "node"`. So that travis will automatically choose the most suitable version.

on line 9 change `script: npm test` to:
```
script:
  - npm test
```

add postgres services since postgres database has been setup:
```
services:
  - postgresql
```
take a look at this **[Travis File](https://github.com/UggoPrince/AutoMart/blob/develop/.travis.yml)** and see what I'm saying, especially from line 1 to 13, 19 to 20.

**Comment by voke1** [comment]
Thanks, @UggoPrince I think you're correct.

**Comment by debelistic** [comment]
> you don't need to include `sudo: false`
> 
> where you have ` - "stable"`, replace it with ` - "node"`. So that travis will automatically choose the most suitable version.
> 
> on line 9 change `script: npm test` to:
> 
> ```
> script:
>   - npm test
> ```
> 
> add postgres services since postgres database has been setup:
> 
> ```
> services:
>   - postgresql
> ```
> 
> take a look at this **[Travis File](https://github.com/UggoPrince/AutoMart/blob/develop/.travis.yml)** and see what I'm saying, especially from line 1 to 13, 19 to 20.

If the purpose is to run stable version of node. Using stable would do just fine. Since you declared your language as node_js. Stable means it will use them most stable version of nodejs to run the app. You can see travis doc for more on this

**Comment by debelistic** [comment]
If it is the latest stable version of nodejs, travis will always support. To have a stable version takes a longer time which would have allowed for travis to keep up. We should be concerned about node version when it comes to deployment.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+sudo: false
```
Comment (anneKay) [inline .travis.yml]: This is the default in travis, you can go ahead and remove this

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - "stable"
+cache:
+  directories:
+    - "node_modules"
+
+install: 
+  - npm install
+
+script: 
+  - npm test
```
Comment (anneKay) [inline .travis.yml:18]: We should have a `before_script:`  here which would setup our test database, before running `npm test`

Comment (voke1) [reply .travis.yml:18]: Done

**Review by anneKay — APPROVED** [review]

**Pull request merged**